### PR TITLE
Update stats and scenario testing

### PR DESCRIPTION
**Pull request by NicolasDP**
- [ ] missing: update the open API doc

### CODE REVIEW

File: jormungandr-scenario-tests/src/main.rs [hunk 1]
```diff
@@ -133,3 +133,67 @@ pub fn scenario_1(mut context: Context<ChaChaRng>) {
 
     controller.finalize();
 }
+
+pub fn scenario_2(mut context: Context<ChaChaRng>) {
```
Comment (mzabaluev) [inline jormungandr-scenario-tests/src/main.rs]: Could the name be more informative as to what the scenario tests?

File: jormungandr-scenario-tests/src/main.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ fn main() {
 
     introduction(&context);
 
-    scenario_1(context.derive());
+    scenario_2(context.derive());
```
Comment (mzabaluev) [inline jormungandr-scenario-tests/src/main.rs]: Has `scenario_1` fallen into dead code then?

File: jormungandr-scenario-tests/src/scenario/controller.rs [hunk 3]
```diff
@@ -21,7 +22,7 @@ pub struct ControllerBuilder {
 }
 
 pub struct Controller {
-    settings: Arc<Settings>,
+    settings: Settings,
```
Comment (mzabaluev) [inline jormungandr-scenario-tests/src/scenario/controller.rs]: Good.

File: jormungandr-scenario-tests/src/wallet/account.rs [hunk 4]
```diff
@@ -46,10 +46,19 @@ impl Wallet {
...
+        writeln!(w, "{}", self.signing_key().to_bech32_str())
+    }
+
     pub fn address(&self, discrimination: Discrimination) -> Address {
         self.identifier().to_address(discrimination).into()
     }
 
+    pub fn increment_counter(&mut self) {
+        let v : u32 = self.internal_counter.into();
+        self.internal_counter = account::SpendingCounter::from(v + 1);
```
Comment (mzabaluev) [inline jormungandr-scenario-tests/src/wallet/account.rs]: `SpendingCounter` might benefit from an impl of `AddAssign` to simplify this.

File: jormungandr-scenario-tests/src/main.rs (hunk 1 above)
Comment (NicolasDP) [reply jormungandr-scenario-tests/src/main.rs]: nop. they will be cleaned up when I have a better control of the different scenario and their execution

File: jormungandr-scenario-tests/src/main.rs (hunk 2 above)
Comment (NicolasDP) [reply jormungandr-scenario-tests/src/main.rs]: soon

File: jormungandr-scenario-tests/src/wallet/account.rs (hunk 4 above)
Comment (NicolasDP) [reply jormungandr-scenario-tests/src/wallet/account.rs]: agree

**Pull request merged**